doc/dev/custom_devices.rst

```diff
@@ -196,137 +196,135 @@ headers and fields are generally required, unless stated otherwise.
+    # CONDS: constraints on the support for each measurement process.
+    #        - "analytic" or "finiteshots": if this measurement is only supported
+    #          in either analytic execution or with shots, respectively.
+    #
+    [measurement_processes]
+
+    ExpectationMP = { }
+    SampleMP = { }
+    CountsMP = { conditions = ["finiteshots"] }
+    StateMP = { conditions = ["analytic"] }
+
+    # Additional support that the device may provide that informs the compilation
+    # process. All accepted fields and their default values are listed below.
+    [compilation]
+
+    # Whether the device is compatible with qjit.
+    qjit_compatible = false
+
+    # Whether the device requires run time generation of the quantum circuit.
+    runtime_code_generation = false
+
+    # Whether the device supports allocating and releasing qubits during execution.
+    dynamic_qubit_management = false
+
+    # Whether simultaneous measurements on overlapping wires is supported.
+    overlapping_observables = true
+
+    # Whether simultaneous measurements of non-commuting observables is supported.
+    # If false, a circuit with multiple non-commuting measurements will have to be
+    # split into multiple executions for each subset of commuting measurements.
+    non_commuting_observables = false
+
+    # Whether the device supports initial state preparation.
+    initial_state_prep = false
+
+    # The methods of handling mid-circuit measurements that the device supports,
+    # e.g., "one-shot", "tree-traversal", "device", etc. An empty list indicates
+    # that the device does not support mid-circuit measurements.
+    supported_mcm_methods = [ ]
+
+This TOML file is used by both Catalyst frontend and PennyLane. Regular circuit execution is
+performed by your implementation of ``Device.execute``, whereas for a QJIT-compiled workflow,
+execution is performed by the ``QuantumDevice``. The TOML file should declare the capabilities
+of the two execution interfaces. If one of the interfaces have additional support that the other
+does not have, include them in a separate section:
+
+.. code-block:: toml
+
+    # Gates supported by the Python implementation of Device.execute but not by the QuantumDevice.
+    [pennylane.operators.gates]
+
+    MultiControlledX  = { }
+
+    # Observables supported by the QuantumDevice but not by your implementation of Device.execute.
+    [qjit.operators.observables]
+
+    Sum = { }
+
+Additionally, any runtime parameters to be passed to the ``QuantumDevice`` upon initialization
+should be specified in a dictionary class property ``device_kwargs`` that links keyword arguments
+of the ``QuantumDevice`` constructor to attributes of the ``qml.device.Device`` implementation.
+For example:
+
+.. code-block:: python
+
+    class CustomDevice(qml.devices.Device):
+        """Custom Device"""
+
+        config_filepath = pathlib.Path("absolute/path/to/configuration/file.toml")
+        device_kwargs = {"arg1": "field1"}
```

Comment:
Will the values change after device initialization `qml.device(...)` but before compiling a circuit with qjit?

Reply:
Good question. Only the device developer could answer that for sure, but I think typically these would static options passed to the device constructor by the user anyway 🤔

Reply:
I say we keep it like this for now because this is how it's always been. I'm also thinking of situations when some of these parameters might not exist. For example, the user would specify ``device_kwargs={'arg1': '_prop1'}``, but ``_prop1`` may or may not exist, and the device is saying only pass it to the backend device if it exists (which is what Catalyst is currently doing I believe)?

Reply:
>For example, the user would specify device_kwargs={'arg1': '_prop1'}, but _prop1 may or may not exist, and the device is saying only pass it to the backend device if it exists (which is what Catalyst is currently doing I believe)?

Oh I didn't mean that users could customize this field, what I meant was nothing changes for the user:
```py
dev = qml.device("my.device", wires=n, do_something=True, special_param="this")
```
and then the device developer would write their constructor as follows:
```py
   def __init__(self, wires, do_something=False, special_param=""):
      # regular constructor stuff
      
      # fully controlled by device developer, but already containing the eventual values for the parameters
      self.device_kwargs = {'cpp_do_something' = do_something, 'cpp_special_param' = special_param}
```
Then, Catalyst just fetches `dev.device_kwargs`, serializes it and sends it down to the runtime/C++ device.

Whereas right now, it's more roundabout where the device developer specifies instance attribute names as a class attribute
```py
class MyDevice:
   device_kwargs = {'cpp_do_something' = 'do_something', 'cpp_special_param' = 'special_param'}
```
and the constructor stores those accordingly
```py
   def __init__(self, wires, do_something=False, special_param=""):
      # regular constructor stuff
      
      self.do_something = do_something
      self.special_param = special_param
```
and then Catalyst needs to go through the class attribute `device_kwargs` to fetch the instance attributes and then serialize the dictionary and send it down.

Reply:
Ah I see what you mean. Yes, I will make the update.

Reply:
Done